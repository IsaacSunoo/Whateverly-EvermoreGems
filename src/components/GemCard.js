import React from 'react';

function GemCard(props) {
    return(
      <div className="gem-cards">
        <img className="gem-pic" src= {props.gemImg}/>
        <article className="gem-info">
          <h5 className="gem-name">
            {props.gemName}
          </h5>
          <h6 className="gem-family">
            {props.gemFam}
          </h6>
        </article>
        <button onClick={() => props.selectGem(props.gemId)} className="select-btn">Select</button>
      </div>
    )
}

export default GemCard;

// we add an anon func to button so it dosn't run on button creation //