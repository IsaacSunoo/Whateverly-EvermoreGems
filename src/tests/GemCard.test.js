import React from 'react';
import GemCard from '../components/GemCard.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const selectGemMock = jest.fn();

const mockGem = [{
  "name": "Aquamarine",
  "id": 1,
  "color": ["Blue"],
  "family": "Beryl",
  "hardness": [7.5, 8],
  "image": "https://i.imgur.com/mupSKCl.jpg"
}]



describe('GemCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <GemCard
      selectGem={selectGemMock}
        gemId={mockGem.id}
        gemName ={mockGem.name}
        gemFam = {mockGem.family}/>
    )
  })
  
  it('should match the snapshot with all data passed in' , () =>{
    expect(wrapper).toMatchSnapshot();
  })
});



