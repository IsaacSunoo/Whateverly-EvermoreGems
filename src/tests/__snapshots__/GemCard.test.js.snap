// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GemCard should match the snapshot with all data passed in 1`] = `
<div
  className="gem-cards"
>
  <img
    className="gem-pic"
  />
  <article
    className="gem-info"
  >
    <h5
      className="gem-name"
    />
    <h6
      className="gem-family"
    />
  </article>
  <button
    className="select-btn"
    onClick={[Function]}
  >
    Select
  </button>
</div>
`;
